src/SFML/System/Android/ResourceStream.cpp

```diff
@@ -41,62 +41,41 @@ ResourceStream::ResourceStream(const std::filesystem::path& filename)
     ActivityStates&       states = getActivity();
     const std::lock_guard lock(states.mutex);
     m_file.reset(AAssetManager_open(states.activity->assetManager, filename.c_str(), AASSET_MODE_UNKNOWN));
+    assert(m_file && "Failed to initialize ResourceStream file");
 }
 
 
 ////////////////////////////////////////////////////////////
-std::int64_t ResourceStream::read(void* data, std::int64_t size)
+std::optional<std::size_t> ResourceStream::read(void* data, std::size_t size)
 {
-    if (m_file)
-    {
-        return AAsset_read(m_file.get(), data, static_cast<std::size_t>(size));
-    }
-    else
-    {
-        return -1;
-    }
+    const auto numBytesRead = AAsset_read(m_file.get(), data, size);
+    if (numBytesRead < 0)
+        return std::nullopt;
+    return numBytesRead;
 }
 
 
 ////////////////////////////////////////////////////////////
-std::int64_t ResourceStream::seek(std::int64_t position)
+std::optional<std::size_t> ResourceStream::seek(std::size_t position)
 {
-    if (m_file)
-    {
-        return AAsset_seek(m_file.get(), static_cast<off_t>(position), SEEK_SET);
-    }
-    else
-    {
-        return -1;
-    }
+    const auto newPosition = AAsset_seek(m_file.get(), static_cast<off_t>(position), SEEK_SET);
+    if (newPosition < 0)
+        return std::nullopt;
+    return newPosition;
 }
 
 
 ////////////////////////////////////////////////////////////
-std::int64_t ResourceStream::tell()
+std::optional<std::size_t> ResourceStream::tell()
 {
-    if (m_file)
-    {
-        return getSize() - AAsset_getRemainingLength(m_file.get());
-    }
-    else
-    {
-        return -1;
-    }
+    return getSize().value() - static_cast<std::size_t>(AAsset_getRemainingLength(m_file.get()));
```

Comment:
Does this have the same semantics as before? Wouldn't this be closer?

```cpp
if (!m_file) { return std::nullopt; }
return getSize().value() - static_cast<std::size_t>(AAsset_getRemainingLength(m_file.get()));

Reply:
`m_file` cannot be null so we're free to omit that check.